client/netdb.c

```diff
@@ -266,6 +275,41 @@ static void *monitor_thread(void *arg)
 	return 0;
 }
 
+static void *refresh_thread(void *arg)
+{
+	while (!g_xdag_sync_on) {
+		sleep(1);
+	}
+	
+	for (;;) {
+		time_t t = time(0);
+		
+		xdag_mess("try to refresh white-list...");
+		
+		char *resp = http_get(WHITE_URL);
+		if(!resp) {
+			goto nextloop;
+		}
+		pthread_mutex_lock(&g_white_list_mutex);
+		FILE *f = xdag_open_file(DATABASEWHITE, "w");
+		if(f) {
+			fwrite(resp, 1, strlen(resp), f);
+			fclose(f);
+		}
+		pthread_mutex_unlock(&g_white_list_mutex);
+		
+		xdag_info(resp);
+		free(resp);
+		
+nextloop:
+		while (time(0) - t < 67) {
```

Comment:
Each 67 seconds?

Reply:
Yep, just keep the same interval as read white-list.

Reply:
Better to use 15 minutes, 67 seconds is too short.

Reply:
Agree, I was thinking to suggest even one hour, but I decided it is just my opinion.